lib/net/dns.rb

```diff
@@ -1,33 +1,59 @@
 require "resolv"
 require "timeout"
+require 'ipaddr'
 
 module Net
   module DNS
-    autoload :ARecord,   "net/dns/a_record.rb"
-    autoload :PTRRecord, "net/dns/ptr_record.rb"
+    autoload :ForwardRecord, "net/dns/forward_record.rb"
+    autoload :ARecord,       "net/dns/a_record.rb"
+    autoload :AAAARecord,    "net/dns/aaaa_record.rb"
+    autoload :ReverseRecord, "net/dns/reverse_record.rb"
+    autoload :PTR4Record,    "net/dns/ptr4_record.rb"
+    autoload :PTR6Record,    "net/dns/ptr6_record.rb"
 
     # Looks up the IP or MAC address. Handles the conversion of a DNS miss
     # exception into nil
     # [+query+]: IP or hostname
-    # Returns: a new DNS record object, A or PTR accordingly
+    # Returns: a new DNS record object, A, AAAA, PTR4 or PTR6 accordingly
     # We query DNS directly, as its faster then to query our own proxy.
-    def self.lookup(query, proxy, resolver = Resolv::DNS.new)
+    def self.lookup(query, options = {})
+      return nil unless query.present?
+
+      proxy = options.fetch(:proxy)
+      resolver = options.fetch(:resolver, Resolv::DNS.new)
+      ipversion = options.fetch(:ipversion, 4)
+
       Timeout.timeout(Setting[:dns_conflict_timeout]) do
-        if (query =~ Validations::IP_REGEXP)
+        ip = IPAddr.new(query) rescue false
+        if ip && ip.ipv4?
           n = resolver.getname(query).to_s
```

Comment:
If you could rename these 3 variables to something more meaningful that'd be great :smile: